### PR TITLE
KRun: add `--search-final` and `--no-pattern` search options

### PR DESCRIPTION
**Pull request by nishantjr**

### CODE REVIEW

**Comment by nishantjr** [comment]
Note that the functions `run` and `run_kore` that return a `CTerm` cannot support this functionality.
They would need to be changed to return a `List[CTerm]` instead.

File: src/tests/unit/ktool/test_krun.py [hunk 1]
```diff
@@ -0,0 +1,42 @@
+from pathlib import Path
+
+from pyk.ktool.krun import KRunOutput, _build_arg_list
+
+
+def test_all_args_1() -> None:
```
Comment (tothtamas28) [inline src/tests/unit/ktool/test_krun.py:6]:
```suggestion
def test_all_args() -> None:
```

Comment (tothtamas28) [reply src/tests/unit/ktool/test_krun.py:6]: Consider testing each optional argument separately with a parametric test.

File: src/pyk/ktool/krun.py [hunk 2]
```diff
@@ -147,6 +151,8 @@ def run_kore_config(
         expand_macros: bool = False,
         bug_report: Optional[BugReport] = None,
         expect_rc: int = 0,
+        search_final: bool = False,
+        no_pattern: bool = False,
```
Comment (tothtamas28) [inline src/pyk/ktool/krun.py:155]:
```suggestion
        search_final: bool = False,
        no_pattern: bool = False,
        # ---
        bug_report: Optional[BugReport] = None,
        expect_rc: int = 0,
```

Here, `# ---` separates kast CLI parameters from the rest.

File: src/pyk/ktool/krun.py [hunk 3]
```diff
@@ -175,7 +181,13 @@ def _sort(p: Pattern) -> KSort:
...
 
         return self.run_kore_term(
-            term, depth=depth, expand_macros=expand_macros, bug_report=bug_report, expect_rc=expect_rc
+            term,
+            depth=depth,
+            expand_macros=expand_macros,
+            bug_report=bug_report,
+            expect_rc=expect_rc,
+            search_final=search_final,
+            no_pattern=no_pattern,
```
Comment (tothtamas28) [inline src/pyk/ktool/krun.py:190]:
```suggestion
            search_final=search_final,
            no_pattern=no_pattern,
            bug_report=bug_report,
            expect_rc=expect_rc,
```

File: src/pyk/ktool/krun.py [hunk 4]
```diff
@@ -114,6 +114,8 @@ def run_kore_term(
         expand_macros: bool = False,
         bug_report: Optional[BugReport] = None,
         expect_rc: Union[int, Iterable[int]] = 0,
+        search_final: bool = False,
+        no_pattern: bool = False,
```
Comment (tothtamas28) [inline src/pyk/ktool/krun.py:118]:
```suggestion
        search_final: bool = False,
        no_pattern: bool = False,
        bug_report: Optional[BugReport] = None,
        expect_rc: Union[int, Iterable[int]] = 0,
```

File: src/pyk/ktool/krun.py [hunk 5]
```diff
@@ -130,6 +132,8 @@ def run_kore_term(
                 no_expand_macros=not expand_macros,
                 bug_report=self._bug_report,
                 check=(expect_rc == 0),
+                search_final=search_final,
+                no_pattern=no_pattern,
```
Comment (tothtamas28) [inline src/pyk/ktool/krun.py:136]:
```suggestion
                search_final=search_final,
                no_pattern=no_pattern,
                bug_report=self._bug_report,
                check=(expect_rc == 0),
```

**Review by tothtamas28 — APPROVED** [review]

**Pull request merged**